### PR TITLE
add kitten plugin. tracks hunger, attention and growth

### PR DESCRIPTION
**Pull request by Nachtmerrie**
This plugin helps you track the progress in growth, hunger and attention of your kitten. This is done in runelite, as the progress is not sent to the client. Every tracker is configurable. 

Also added system notifications for when your kittycat is hungry or requires your attention.

I had lots of fun making this. Learned a lot about the game, and had my first time playing around in Java!
Let me know what you think

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/kitten/Felines.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package net.runelite.client.plugins.kitten;
```
Comment (ypperlig) [inline runelite-client/src/main/java/net/runelite/client/plugins/kitten/Felines.java]: Add BSD license to new files

File: runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenAttentionTimer.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package net.runelite.client.plugins.kitten;
```
Comment (ypperlig) [inline runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenAttentionTimer.java]: Add BSD license to new files

File: runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenConfig.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
+package net.runelite.client.plugins.kitten;
```
Comment (ypperlig) [inline runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenConfig.java]: Add BSD license to new files

File: runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenGrowthTimer.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package net.runelite.client.plugins.kitten;
```
Comment (ypperlig) [inline runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenGrowthTimer.java]: Add BSD license to new files

File: runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenHungryTimer.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package net.runelite.client.plugins.kitten;
```
Comment (ypperlig) [inline runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenHungryTimer.java]: Add BSD license to new files

File: runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenAttentionTimer.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+import net.runelite.client.ui.overlay.infobox.Timer;
+import net.runelite.client.plugins.Plugin;
+
+import java.awt.image.BufferedImage;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+
+
+public class kittenAttentionTimer extends Timer
```
Comment (Infinitay) [inline runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenAttentionTimer.java]: Class names should be capital `KittenAttentionTimer`

File: runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenPlugin.java [hunk 7]
```diff
@@ -0,0 +1,568 @@
...
+import net.runelite.client.util.Text;
+
+import javax.inject.Inject;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+
+@PluginDescriptor(
+		name = "Kitten"
```
Comment (Infinitay) [inline runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenPlugin.java]: A suggestion I have is to give it a better name along the lines of "Kitten Tracker" or something like that. If you do end up changing it, I think you also have to change it in your `Config` class too if I'm not mistaken

File: runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenPlugin.java [hunk 8]
```diff
@@ -0,0 +1,568 @@
...
+		// followerID is the first 2 octets converted into decimals
+		followerID = Integer.parseInt(s.substring(0, 16), 2);
+		if (followerID != 0) //Varbit needs to fill up first after logging in
+		{
+			NewFollower();
+		}
+
+	}
+
+	private void NewFollower()
```
Comment (Infinitay) [inline runelite-client/src/main/java/net/runelite/client/plugins/kitten/kittenPlugin.java]: I think you accidentally forgot to lowercase `new`

**Comment by Nachtmerrie** [comment]
Thank you @Infinitay for the kind words. I just PascalCased my classes ^_^

**Comment by Nachtmerrie** [comment]
Closing this PR because of the project redesign

**Pull request closed without merging**